main.go

```diff
@@ -76,6 +77,11 @@ func printMetricsPerMetricGenerator(pathToRules string) {
 	metricsIdentifiers := make(map[string]map[string]struct{})
 	for metric := range processing.Metrics(expressions) {
 		metricMetadata := prom.MetricMetadata(metric)
+		if len(metricMetadata) == 0 && (strings.HasSuffix(metric, "total") || strings.HasSuffix(metric, "bucket")) {
+			metricSplit := strings.Split(metric, "_")
+			sisterMetric := strings.Join(metricSplit[:len(metricSplit)-1], "_")
+			metricMetadata = prom.MetricMetadata(sisterMetric)
+		}
```

Comment:
(nit) a bit shorter :)

```suggestion
		if len(metricMetadata) == 0 {
			// Lookup for the parent metric name if it's a counter, histogram or summary. 
			for _, s := range []string{"_total", "_bucket", "_sum", "_count"} {
				metricMetadata = prom.MetricMetadata(strings.TrimSuffix(metric, s))
				if len(metricMetadata) > 0 {
					break
				}
			}
		}
```

Reply:
Great! Yes, this is wayyy more legible!